### ISSUE TITLE
Pod install Errno::EEXIST - File exists

### ISSUE THREAD

**Issue by davideme** [CocoaPods/CocoaPods#5700]
🌈
* [X] I've read and understood the [*CONTRIBUTING guidelines and have done my best effort to follow](https://github.com/CocoaPods/CocoaPods/blob/master/CONTRIBUTING.md).

# Report

## What did you do?

Run `pod install`

## What did you expect to happen?
Install all pod dependencies correctly.

## What happened instead?
An error occurred.

### Error

```
Errno::EEXIST - File exists - /Users/davide/GitHub/BothamUI/Pods/Target Support Files/Nimble
/Library/Ruby/Gems/2.0.0/gems/cocoapods-1.1.0.beta.1/lib/cocoapods/installer/xcode/pods_project_generator/target_installer.rb:92:in `mkdir'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-1.1.0.beta.1/lib/cocoapods/installer/xcode/pods_project_generator/target_installer.rb:92:in `mkdir'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-1.1.0.beta.1/lib/cocoapods/installer/xcode/pods_project_generator/target_installer.rb:92:in `create_support_files_dir'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-1.1.0.beta.1/lib/cocoapods/installer/xcode/pods_project_generator/pod_target_installer.rb:21:in `block in install!'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-1.1.0.beta.1/lib/cocoapods/user_interface.rb:142:in `message'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-1.1.0.beta.1/lib/cocoapods/installer/xcode/pods_project_generator/pod_target_installer.rb:19:in `install!'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-1.1.0.beta.1/lib/cocoapods/installer/xcode/pods_project_generator.rb:155:in `block (2 levels) in install_libraries'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-1.1.0.beta.1/lib/cocoapods/installer/xcode/pods_project_generator.rb:153:in `each'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-1.1.0.beta.1/lib/cocoapods/installer/xcode/pods_project_generator.rb:153:in `block in install_libraries'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-1.1.0.beta.1/lib/cocoapods/user_interface.rb:142:in `message'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-1.1.0.beta.1/lib/cocoapods/installer/xcode/pods_project_generator.rb:152:in `install_libraries'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-1.1.0.beta.1/lib/cocoapods/installer/xcode/pods_project_generator.rb:64:in `generate!'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-1.1.0.beta.1/lib/cocoapods/installer.rb:181:in `block in generate_pods_project'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-1.1.0.beta.1/lib/cocoapods/user_interface.rb:64:in `section'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-1.1.0.beta.1/lib/cocoapods/installer.rb:180:in `generate_pods_project'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-1.1.0.beta.1/lib/cocoapods/installer.rb:116:in `install!'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-1.1.0.beta.1/lib/cocoapods/command/install.rb:37:in `run'
/Library/Ruby/Gems/2.0.0/gems/claide-1.0.0/lib/claide/command.rb:334:in `run'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-1.1.0.beta.1/lib/cocoapods/command.rb:50:in `run'
/Library/Ruby/Gems/2.0.0/gems/cocoapods-1.1.0.beta.1/bin/pod:55:in `<top (required)>'
/usr/local/bin/pod:23:in `load'
/usr/local/bin/pod:23:in `<main>'
```

### Stack

```
   CocoaPods : 1.1.0.beta.1
        Ruby : ruby 2.0.0p648 (2015-12-16 revision 53162) [universal.x86_64-darwin15]
    RubyGems : 2.0.14.1
        Host : Mac OS X 10.11.6 (15G31)
       Xcode : 7.3.1 (7D1012)
         Git : git version 2.7.4 (Apple Git-66)
Ruby lib dir : /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib
Repositories : bitbucket-karumi-pods - git@bitbucket.org:karumi/karumi-pods.git @ 2b2b7cffdd37285603875df95e357716f998153d
               master - https://github.com/CocoaPods/Specs.git @ 714d2b0fb512735a9d0a04f93f93680b46dede7f
```

### Installation Source

```
Executable Path: /usr/local/bin/pod
```

### Plugins

```
cocoapods-deintegrate : 1.0.0
cocoapods-packager    : 1.3.0
cocoapods-plugins     : 1.0.0
cocoapods-search      : 1.0.0
cocoapods-stats       : 1.0.0
cocoapods-trunk       : 1.0.0
cocoapods-try         : 1.1.0
```

### Podfile

```ruby
workspace 'BothamUI.xcworkspace'
project 'BothamUI.xcodeproj'
platform :ios, '8.0'
use_frameworks!

target 'BothamUIUnitTests' do
    pod 'Nimble'
end

target 'Example' do
        project 'Example/Example.xcodeproj'
    pod 'BothamUI', :path => './'
    pod 'SDWebImage', '~>3.7'
end

target 'ExampleTests' do
    project 'Example/Example.xcodeproj'
    pod "Nimble"
end

target 'ExampleAcceptanceTests' do
        project 'Example/Example.xcodeproj'
    pod 'BothamUI', :path => "./"
    pod 'Nimble'
    pod 'KIF', '~> 3.0', :configurations => ['Debug']
end
```

## Project that demonstrates the issue

https://github.com/Karumi/BothamUI/tree/swift-2.3

**Comment by segiddins** [comment]
This is possibly a duplicate of https://github.com/CocoaPods/CocoaPods/issues/4837?

**Comment by davideme** [comment]
I don't think so because my project is working with the same Podfile with pod 1.0.1
And there is no target with repeated name.

**Comment by SlaunchaMan** [comment]
I’m getting this as well in IRLSize with the tag [1.0.0-beta1](https://github.com/detroit-labs/IRLSize/releases/tag/1.0.0-beta1).

**Comment by jtdavid** [comment]
Same issue here. I have common dependencies for multiple targets and with CocoaPods 1.0.1 it works well. It only happens with cocoapods-1.1.0.beta.1...

**Comment by Coeur** [comment]
@segiddins, maybe more like a duplicate of #5661. It works fine with CocoaPods 1.0.1, but we have the "Errno::EEXIST - File exists" with CocoaPods 1.1.0 beta 1 and Xcode 8 beta 4.

**Comment by Coeur** [comment]
Nevermind: in my specific case, adding `inherit! :search_paths` fixed the issue, so I was facing something different.

**Comment by benasher44** [comment]
@davideme somehow your test ExampleTests and ExampleAcceptanceTests targets have no swift version set. If you can re-add those targets or otherwise get those targets to have swift versions, your issue will go away. @segiddins should we raise an `Informative` in this scenario? We use swift version in `PodVariant` to help de-duplicate targets.

**Comment by segiddins** [comment]
@benasher44 I guess we probably should

**Comment by benasher44** [comment]
Actually this might be a bug with how we get the swift version to store in a `TargetDefinition`

**Comment by benasher44** [comment]
Verified this is fixed with #5760

**Comment by davideme** [comment]
Tested with CocoaPods-49b4b5afe415 and Confirmed as fixed
